tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -676,28 +675,51 @@ def get_iterator(self, iterator_id):
     def iterator_completed(self, iterator_id):
       del self._iterators[iterator_id]
 
+
   @staticmethod
-  def from_generator(generator, output_types, output_shapes=None, args=None):
+  @deprecation.deprecated_args(None, "Use output_spec instead",
+                               "output_types", "output_shapes")
+  # TODO(lithuak): Make output_spec a required argument once output_types
+  # and output_shapes are removed.
+  def from_generator(generator,
+                     output_types=None,
+                     output_shapes=None,
+                     args=None,
+                     output_spec=None):
```

Comment:
I think it would be better to name the new argument `output_signature` rather than `output_spec`.  In particular, we've been using "spec" to refer to a single TypeSpec, and "signature" to refer to a possibly nested structure of TypeSpecs.  E.g., `tf.function` has an `input_signature` argument, and `tf.map_fn` has recently been updated to have a `fn_output_signature` argument.  In this case, the argument can take a complete signature (not just a single TypeSpec), so I believe it should be named `output_signature`.

Reply:
@edloper The name was proposed by @jsimsa I think it's better if you two decide on this together ^^

Reply:
Neither `output_spec` nor `output_signature` is fully consistent. tf.data has `tf.data.Dataset.element_spec` which is a structure of `TypeSpec`s. I would slightly lean towards `output_spec` as that does not require any changes and is consistent within the tf.data API surface. @edloper do you feel strongly the other way?

Reply:
@jsimsa @edloper I second not changing it. It's not only the public API surface but also the internal APIs (e.g. Structure module) that use the "_spec" suffix all over the code. After reading this code over the last weeks my feeling is that "_signature" would be a bit misleading.

Reply:
OK, let's leave it as is for consistency within tf.data.